compiler/src/main/java/com/squareup/anvil/compiler/codegen/BindingModuleGenerator.kt

```diff
@@ -294,23 +293,21 @@ internal class BindingModuleGenerator(
     }
   }
 
-  private fun findContributedBindingClasses(projectFiles: Collection<KtFile>) {
+  private fun findContributedBindingClasses(
+    module: ModuleDescriptor,
+    projectFiles: Collection<KtFile>
+  ) {
     contributedBindingClasses += projectFiles
       .filter {
-        it.packageFqName.asString()
-          .startsWith(HINT_BINDING_PACKAGE_PREFIX)
+        it.packageFqName.asString().startsWith(HINT_BINDING_PACKAGE_PREFIX)
       }
       .flatMap {
-        it.findChildrenByClass(KtProperty::class.java)
-          .toList()
+        it.findChildrenByClass(KtProperty::class.java).toList()
       }
       .mapNotNull { ktProperty ->
-        (
-          (ktProperty.initializer as? KtClassLiteralExpression)
-            ?.firstChild as? KtDotQualifiedExpression
-          )
-          ?.text
-          ?.let { FqName(it) }
```

Comment:
This wasn't great, it was pretty much hardcoded that a fully qualified name is used for `val prop = abc.def.Class::class`.